src/OpenConext/ProfileBundle/Security/Authentication/Provider/SamlProvider.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * Copyright 2015 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\ProfileBundle\Security\Authentication\Provider;
+
+use OpenConext\ProfileBundle\Security\Authentication\Entity\User;
+use OpenConext\ProfileBundle\Security\Authentication\Token\SamlToken;
+use Surfnet\SamlBundle\SAML2\Attribute\AttributeDictionary;
+use Symfony\Component\Security\Core\Authentication\Provider\AuthenticationProviderInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+class SamlProvider implements AuthenticationProviderInterface
+{
+    /**
+     * @var \Surfnet\SamlBundle\SAML2\Attribute\AttributeDictionary
+     */
+    private $attributeDictionary;
+
+    public function __construct(
+        AttributeDictionary $attributeDictionary
+    ) {
+        $this->attributeDictionary = $attributeDictionary;
+    }
+
+    /**
+     * @param  TokenInterface $token
+     * @return TokenInterface|void
+     */
+    public function authenticate(TokenInterface $token)
```

Comment:
Is this method part of the interface? Then docblocks are discouraged (read: remove :wink:).